### PR TITLE
Prepare 1.0.1

### PR DESCRIPTION
**Pull request by viktorklang**
WIP

- [ ] Ship 1.0.1-RC1
- [x] Update RELEASE-NOTES.md
- [ ] Write release announcement

### CODE REVIEW

**Comment by viktorklang** [comment]
You mean like https://github.com/reactive-streams/reactive-streams-jvm/milestone/10 ?

**Comment by ktoso** [comment]
Yeah, that's how I meant, I missed that milestones got assigned to most, thanks :)

**Comment by viktorklang** [comment]
@ktoso Yeah, and I've just pushed a draft of release notes. Would you be able to summarize the improvements to the TCK?

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -1,3 +1,135 @@
...
+## Highlights:
+
+- Specification
+  + A new [Glossary](https://github.com/reactive-streams/reactive-streams-jvm/blob/v1.0.1-RC1/README.md#glossary) section
+  + Description of the intent behind every single rule
+  + No breaking semantical changes
+- Interfaces
+  + No changes
+  + Improved Javadoc
+- TCK
```
Comment (viktorklang) [inline RELEASE-NOTES.md]: @akarnokd @ktoso Could you please take a stab at listing the improvements to the TCK? Thank you!
(It's fine to add it as comments here and I'll incorporate it into the release notes.)

/cc @reactive-streams/contributors

File: RELEASE-NOTES.md [hunk 2]
```diff
@@ -1,3 +1,146 @@
...
+  + No breaking semantical changes
+  + Multiple rule [clarifications](#specification-clarifications)
+- Interfaces
+  + No changes
+  + Improved JavaDoc
+- TCK
+  + Improved coverage
+  + Improved JavaDoc
+  + Multiple test [alterations](#tck-alterations)
+- New contributors since 1.0.0
```
Comment (ktoso) [inline RELEASE-NOTES.md]: weird style to be honest, usually one lists contributors who contributed to a given milestone/release, not only the new ones?

File: RELEASE-NOTES.md [hunk 3]
```diff
@@ -1,3 +1,146 @@
...
+
+**Comment: Rule simplified by defining `demand` in the glossary, and footnote was reworked into the Intent-section of the rule.**
+
+---
+
+## TCK alterations
+
+- Fixed potential resource leaks in partially consuming Publisher tests
+- Fixed potential resource leaks in partially emitting Subscriber tests
+- Renamed `untested_spec305_cancelMustNotSynchronouslyPerformHeavyCompuatation` to `untested_spec305_cancelMustNotSynchronouslyPerformHeavyComputation` (typo)
```
Comment (ktoso) [inline RELEASE-NOTES.md]: Link to https://github.com/reactive-streams/reactive-streams-jvm/pull/306

File: RELEASE-NOTES.md [hunk 4]
```diff
@@ -1,3 +1,146 @@
...
+
+**1.0.0:** A Subscription MUST support an unbounded number of calls to request and MUST support a demand (sum requested - sum delivered) up to 2^63-1 (java.lang.Long.MAX_VALUE). A demand equal or greater than 2^63-1 (java.lang.Long.MAX_VALUE) MAY be considered by the Publisher as “effectively unbounded”[3].
+
+**1.0.1:** A Subscription MUST support an unbounded number of calls to request and MUST support a demand up to 2^63-1 (java.lang.Long.MAX_VALUE). A demand equal or greater than 2^63-1 (java.lang.Long.MAX_VALUE) MAY be considered by the Publisher as “effectively unbounded”.
+
+**Comment: Rule simplified by defining `demand` in the glossary, and footnote was reworked into the Intent-section of the rule.**
+
+---
+
+## TCK alterations
```
Comment (ktoso) [inline RELEASE-NOTES.md]: Add:

- Allow configuring separate timeout for "no events during N time", allowing for more aggressive timeouts in the rest of the test suite if required https://github.com/reactive-streams/reactive-streams-jvm/pull/314
- New test verifying Rule 2.10, in which subscriber must be prepared to receive onError signal without having signaled `request` before https://github.com/reactive-streams/reactive-streams-jvm/pull/374

those are more interesting than the remaining "just a fix" ones :)

File: RELEASE-NOTES.md [hunk 5]
```diff
@@ -1,3 +1,146 @@
...
+
+- Fixed potential resource leaks in partially consuming Publisher tests
+- Fixed potential resource leaks in partially emitting Subscriber tests
```
Comment (ktoso) [inline RELEASE-NOTES.md]: Link to https://github.com/reactive-streams/reactive-streams-jvm/pull/372  and https://github.com/reactive-streams/reactive-streams-jvm/pull/373

File: RELEASE-NOTES.md [hunk 6]
```diff
@@ -1,3 +1,146 @@
...
+
+**1.0.1:** A Subscription MUST support an unbounded number of calls to request and MUST support a demand up to 2^63-1 (java.lang.Long.MAX_VALUE). A demand equal or greater than 2^63-1 (java.lang.Long.MAX_VALUE) MAY be considered by the Publisher as “effectively unbounded”.
+
+**Comment: Rule simplified by defining `demand` in the glossary, and footnote was reworked into the Intent-section of the rule.**
+
+---
+
+## TCK alterations
+
+- Fixed potential resource leaks in partially consuming Publisher tests
```
Comment (ktoso) [inline RELEASE-NOTES.md]: Link to https://github.com/reactive-streams/reactive-streams-jvm/pull/375

File: RELEASE-NOTES.md (hunk 2 above)
Comment (viktorklang) [inline RELEASE-NOTES.md]: Yeah, you're right, I'll list all, and then annotate the new ones :)

File: RELEASE-NOTES.md (hunk 3 above)
Comment (viktorklang) [inline RELEASE-NOTES.md]: 👍

File: RELEASE-NOTES.md (hunk 4 above)
Comment (viktorklang) [inline RELEASE-NOTES.md]: 👍

File: RELEASE-NOTES.md (hunk 5 above)
Comment (viktorklang) [inline RELEASE-NOTES.md]: 👍

File: RELEASE-NOTES.md (hunk 6 above)
Comment (viktorklang) [inline RELEASE-NOTES.md]: 👍

File: RELEASE-NOTES.md (hunk 2 above)
Comment (ktoso) [inline RELEASE-NOTES.md]: :+1:

File: RELEASE-NOTES.md [hunk 7]
```diff
@@ -1,3 +1,148 @@
...
+
+**Comment: Rule simplified by defining `demand` in the glossary, and footnote was reworked into the Intent-section of the rule.**
+
+---
+
+## TCK alterations
+
+- Fixed potential resource leaks in partially consuming Publisher tests (#375)
+- Fixed potential resource leaks in partially emitting Subscriber tests (#372, #373)
+- Renamed `untested_spec305_cancelMustNotSynchronouslyPerformHeavyCompuatation` to `untested_spec305_cancelMustNotSynchronouslyPerformHeavyComputation` (#306)
```
Comment (ktoso) [inline RELEASE-NOTES.md]: Have the issue numbers be links?

**Comment by viktorklang** [comment]
@reactive-streams/contributors Please review this PR, I think we're ready to ship 1.0.1-RC1 (and have the implementations make sure that the new TCK doesn't cause any havoc).

**Comment by DougLea** [comment]
looks good to me

**Comment by viktorklang** [comment]
@reactive-streams/contributors @rkuhn @smaldini Who wants to do  the honors of publishing the RC1?

**Comment by viktorklang** [comment]
@reactive-streams/contributors Merging this into master and tagging. Publish will happen as soon as we have someone who wants to do the honors :)

**Comment by rkuhn** [comment]
I can probably do it later today, need to finish some other stuff first, though.

**Comment by viktorklang** [comment]
@rkuhn That'd be much welcome. Let me know if you want to coordinate. I'm merging this to master and tagging it.

**Comment by viktorklang** [comment]
@rkuhn merged and tagged

**Comment by ktoso** [comment]
Yay, thanks Roland. Ping if you'll want a sanity check of the artifact

**Comment by rkuhn** [comment]
Manually applying https://github.com/reactive-streams/reactive-streams-jvm/pull/220/commits/aaac86ece814e3f63003c105e1c16053c4c2280a I have signed and uploaded to https://oss.sonatype.org/content/repositories/orgreactivestreams-1030, please let me know if everything arrived okay.

**Comment by viktorklang** [comment]
@rkuhn LGTM!

**Comment by akarnokd** [comment]
RxJava 2.1.2 snapshot passes this TCK locally (107 casses).

**Comment by ktoso** [comment]
Akka Streams 2.5.3 all green: `Passed: Total 534, Failed 0, Errors 0, Passed 534, Ignored 424` 🎉

**Comment by viktorklang** [comment]
:shipit: ?

**Comment by rkuhn** [comment]
Shipped it :-)

**Comment by viktorklang** [comment]
Thanks @rkuhn, @ktoso, and all the @reactive-streams/contributors!

Now we'll give people a week or two to verify that the TCK works as expected and then ship the final version.

**Comment by viktorklang** [comment]
@rkuhn I'ev also pushed the correct tag now: https://github.com/reactive-streams/reactive-streams-jvm/tree/v1.0.1-RC1

**Pull request merged**